tc/core/polyhedral/body.h

```diff
@@ -26,19 +26,31 @@ namespace polyhedral {
 struct Body {
   Body() = default;
   Body(isl::space paramSpace) {
-    writes = reads = isl::union_map::empty(paramSpace);
+    reductions = writes = reads = isl::union_map::empty(paramSpace);
   }
 
   // Specialize to the given context.
   void specialize(isl::set context) {
     reads = reads.intersect_params(context);
     writes = writes.intersect_params(context);
+    reductions = reductions.intersect_params(context);
   }
 
   // Union maps describing the reads and writes done. Uses the ids in
   // the schedule tree to denote the containing Stmt, and tags each
   // access with a unique reference id of the form __tc_ref_N.
   isl::union_map reads, writes;
+
+  // A function on reduction update statement instances that partitions them
+  // into individual reductions.
+  // Each reduction involves a single a single tensor element,
+  // meaning that the partition of statement instances based
+  // on the reductions forms a refinement of the partition based
+  // on the element modified by the statement.
```

Comment:
This comment confused me more than helped.  Can we have the spaces spelled out, maybe with an example?

Reply:
> This comment confused me more than helped.  Can we have the spaces spelled out
, maybe with an example?

Updated.

skimo

Reply:
thanks, isl notation helps a lot